cmake/hipsycl-config.cmake.in

```diff
@@ -48,40 +48,66 @@ if(HIPSYCL_ROCM_BACKEND_AVAILABLE)
   endif()
 endif()
 
+set(HIPSYCL_SYCLCC_EXTRA_ARGS "${HIPSYCL_SYCLCC_EXTRA_ARGS}" CACHE STRING "Arguments to pass through directly to syclcc.")
+
+set(HIPSYCL_TARGETS "${HIPSYCL_TARGETS}" CACHE STRING "The platforms and architectures hipSYCL should target, in the format <platform1>[:arch1[,arch2][..,archN]][;<platform2>[:arch1][...]][..;<platformN>] where platforms are one of ${HIPSYCL_PLATFORMS_STRING}.")
+# IF HIPSYCL_TARGETS has not been explicitly set by the user, first tro to find the corresponding environment variable.
+# If found, takes precedence over deprecated HIPSYCL_PLATFORM and HIPSYCL_GPU_ARCH
+# IF not found, fallback to deprecated HIPSYCL_PLATFORM and HIPSYCL_GPU_ARCH logic
+if(NOT HIPSYCL_TARGETS)
+  set(HIPYSCL_TARGETS_ENV $ENV{HIPSYCL_TARGETS})
+  if(HIPSYCL_TARGETS_ENV)
+    message("Found HIPYSCL_TARGETS from environment: ${HIPSYCL_TARGETS_ENV}")
+    set(HIPSYCL_TARGETS "${HIPSYCL_TARGETS_ENV}")
+  endif()
+endif()
+if(HIPSYCL_TARGETS)
+  # Canonicalize all targets
+  set (HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS}")
+  string(REGEX REPLACE "cpu|host|hipcpu|openmp|omp" "omp" HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS_CANONICAL}")
+  string(REGEX REPLACE "cuda|nvidia" "cuda" HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS_CANONICAL}")
+  string(REGEX REPLACE "rocm|amd|hip|hcc" "hip" HIPSYCL_TARGETS_CANONICAL "${HIPSYCL_TARGETS_CANONICAL}")
```

Comment:
I don't think we need to do that - `syclcc` doesn't accept the old synonyms for `--hipsycl-targets`, and I suspect it would be more intuitive if cmake behaved analogously. Then we can also deprecate the synonyms together with the old `--hipsycl-platform` mechanism in the future.